esmf_regrid/tests/unit/experimental/unstructured_scheme/test_GridToMeshESMFRegridder.py

```diff
@@ -196,3 +200,34 @@ def test_mask_handling():
     assert ma.allclose(expected_0, result_0.data)
     assert ma.allclose(expected_05, result_05.data)
     assert ma.allclose(expected_1, result_1.data)
+
+
+def test_laziness():
+    """Test that regridding is lazy when source data is lazy."""
+    n_lons = 12
+    n_lats = 10
+    h = 4
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+
+    mesh = _gridlike_mesh(n_lons, n_lats)
+
+    src_data = np.arange(n_lats * n_lons * h).reshape([n_lats, n_lons, h])
+    src_data = da.from_array(src_data, chunks=[3, 5, 2])
+    src = Cube(src_data)
+    grid = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+    src.add_dim_coord(grid.coord("latitude"), 0)
+    src.add_dim_coord(grid.coord("longitude"), 1)
+
+    mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+    tgt_data = np.zeros([n_lats * n_lons])
+    tgt = Cube(tgt_data)
+    tgt.add_aux_coord(mesh_coord_x, 0)
+    tgt.add_aux_coord(mesh_coord_y, 0)
```

Comment:
Could any of this be refactored into a pytest fixture?

Reply:
I think in general there is a lot that could be refactored for all of the tests in this directory. One of the things I've also wanted to tidy up is where helper functions (which may become fixtures) are imported from, given they are used in multiple modules. The way things are at the moment is that they exist in the test module that they were originally used in, but now we're in a situation where some helper functions are built from other helper functions that live in different modules and it could all do with a bit of tidying up. I'm unsure how we would have `test_GridToMeshESMFRegridder.py` and `test_MeshToGridESMFRegridder.py` both use the same set of fixtures without either resolving the problem  of where common helper functions come from or having one depend on the other. There may be a case for creating a separate issue for test refactoring.

One more small comment about pytest fixtures, since some helper functions are imported by the asv benchmarks, it might make sense that these remain as ordinary helper functions rather than become pytest fixtures.

Reply:
>some helper functions are built from other helper functions that live in different modules and it could all do with a bit of tidying up

I don't doubt it.

>There may be a case for creating a separate issue for test refactoring.

It sounds like you're saying this would be significant work in its own right? In which case I definitely agree that it should be covered later, and a task should also be raised in our internal JIRA project planner - we need to make sure this gets visibility in planning, else it is likely to remain here causing technical debt for future tasks.

Reply:
>since some helper functions are imported by the asv benchmarks, it might make sense that these remain as ordinary helper functions rather than become pytest fixtures.

Would writing a fixture that simply wraps the function work? From what I've read, fixtures offer a lot of benefit for test structure and clarity, especially for developers who haven't seen this conversation before looking at the tests! So it feels like it would be worth it.

Reply:
raised as issue #116